react/src/components/UserProfileSettingModal.tsx

```diff
@@ -0,0 +1,287 @@
+import React from "react";
+import { Modal, Input, Form, Select, Divider, message, Switch } from "antd";
+import { useTranslation } from "react-i18next";
+import { useWebComponentInfo } from "./DefaultProviders";
+import { passwordPattern } from "./ResetPasswordRequired";
+import { useSuspendedBackendaiClient } from "../hooks";
+import { useTanMutation } from "../hooks/reactQueryAlias";
+
+const UserProfileSettingModal : React.FC = () => {
+  const { t } = useTranslation();
+
+  const [form] = Form.useForm();
+
+  const [messageApi, contextHolder] = message.useMessage();
+  const _showSuccessMessage = (successMessage: string) => {
+    messageApi.open({
+      type: "success",
+      content: successMessage
+    });
+  };
+  const _showErrorMessage = (errorMessage: string) => {
+    messageApi.open({
+      type: "error",
+      content: errorMessage
+    });
+  };
+
+  const baiClient = useSuspendedBackendaiClient();
+  let email = baiClient.email;
+  let full_name = baiClient.full_name;
+  let loggedAcount = baiClient._config.accessKey;
+  let totpSupported = false;
+  let totpActivated = false;
+  let selectOptions: any[] = [];
+  let keyPairInfo = {
+    keypairs: [{
+      access_key: "",
+      secret_key: ""
+    }]
+  };
+
+  const _showTotpRule = async() => {
+    totpSupported = baiClient.supports('2FA') && await baiClient.isManagerSupportingTOTP();
+    if (totpSupported) {
+      const userInfo = await baiClient.user.get( baiClient.email, ['totp_activated']);
+      totpActivated = userInfo.user.totp_activated;
+    };
+  };
+
+  const _showKeypairInfo = async(email: string) => {
+    keyPairInfo = await baiClient.keypair.list(email, ["access_key", "secret_key"], true);
+    if (keyPairInfo.keypairs) {
+      for (let i=0; i < keyPairInfo.keypairs.length; i++) {
+        selectOptions.push({value: keyPairInfo.keypairs[i].secret_key, label: keyPairInfo.keypairs[i].access_key})
+      };
+      for(let i=0; i<selectOptions.length; i++) {
+        if (loggedAcount === selectOptions[i].label) {
+          form.setFieldsValue({
+            access_key: selectOptions[i].label,
+            secret_key: selectOptions[i].value
+          })
+          break;
+        };
+      };
+    };
+  };
+
+  _showTotpRule();
+  _showKeypairInfo(email);
```

Comment:
This function would be called many times. (if you want to test it, print console.log) Rather than calling this function here, just use a variable such as [totpSupported](https://github.com/lablup/backend.ai-webui/blob/51852efbb696c20e94a1055fe6db85e234e85eaf/react/src/components/UserInfoModal.tsx#L42).

Reply:
Naming: And please use `_show<something>()` when you want to display some component such as modal.